Source/WebCore/dom/TrustedType.cpp

```diff
@@ -163,9 +163,15 @@ ExceptionOr<String> trustedTypeCompliantString(TrustedType expectedType, ScriptE
     if (!scriptExecutionContext.settingsValues().trustedTypesEnabled)
         return stringValue;
 
-    CheckedPtr contentSecurityPolicy = scriptExecutionContext.checkedContentSecurityPolicy();
+    auto requireTrustedTypes = false;
 
-    auto requireTrustedTypes = contentSecurityPolicy && contentSecurityPolicy->requireTrustedTypesForSinkGroup("script"_s);
+    if (RefPtr document = dynamicDowncast<Document>(scriptExecutionContext))
+        requireTrustedTypes = document->requiresTrustedTypes();
+    else {
+        CheckedPtr contentSecurityPolicy = scriptExecutionContext.checkedContentSecurityPolicy();
+
+        requireTrustedTypes = contentSecurityPolicy && contentSecurityPolicy->requireTrustedTypesForSinkGroup("script"_s);
+    }
```

Comment:
I'm not sure I understand this change

Reply:
It uses the document's flag when it can, but for other envs (such as Workers) we fallback to the old method of checking the CSP.